src/components/MenuItem.js

```diff
@@ -63,9 +68,9 @@ const MenuItem = ({
                         </Text>
                     </View>
                 </View>
-                {shouldShowRightArrow && (
+                {shouldShowRightIcon && (
```

Comment:
I see several places in the code which passes `iconRight` but not `shouldShowRightIcon` and so that would mean those places never have the `iconRight` displayed?

Example: https://github.com/Expensify/Expensify.cash/blob/bbdd6b6d6b83d7e980dc7f99ffbdab9e6f33100d/src/pages/settings/AppDownloadLinks.js#L21-L39

Reply:
Pls have a look at the iterator over the data
https://github.com/Expensify/Expensify.cash/blob/bbdd6b6d6b83d7e980dc7f99ffbdab9e6f33100d/src/pages/settings/AppDownloadLinks.js#L58

Reply:
Hmm, upon review it is a bit confusing that `shouldShowRightIcon` is defined outside of the `menuItems` array

Reply:
This is because all these items shares the same icon property so it’s better to define it once.

Reply:
In that case, the argument for repeating the Right Icon is a bit weaker, because it would also be defined once for each page...

Reply:
Stepping back, I think there are three cases trying to be handled here:

1. There is no right icon
2. There is a right icon, and it defaults to an arrow
3. There is a right icon, and it is a specific icon

Rather than trying to tackle this in a single component, I think it would be much better to separate this into two components:

- `<MenuItem>` has no right icon
- `<MenuItemWithRightIcon>` has a right icon that defaults to the arrow
- `<MenuItemWithRightIcon icon={foo}>` has a right icon that is a specific icon

I have not looked at all the places in the code where we currently use `<MenuItem>` so I don't know how complex it would be to implement, but I think having the two separate components makes it much easier for someone else to come along and use them without needing to dig into the source to figure out how the props work.

Reply:
@tgolen I would rather prefer this 
```jsx
<MenuItem>
    <MenuItem.RightIcon icon={foo} />
</MenuItem>
``` 
or 

```jsx
<MenuItem>
    <MenuItem.RightIcon>
        <Icon /> // or <SomeComposedComponent />
    <MenuItem.RightIcon>
</MenuItem>
``` 

I really agree that refactoring of building blocks (components) is required to support long term goals.
If we want to move forward with this or any other API for components then I can pull this off as I have some significant experience in design systems.

I am one of the core maintainer of [React Native Elements](https://github.com/react-native-elements/react-native-elements)

Reply:
@pranshuchittora Ah yes, those would work as well! Can you help me understand why you would prefer that pattern over the one I proposed? Is it just personal preference, or is there a specific reason for it?

Reply:
It’s a bit opinionated but makes things easy to maintain and also makes the code readable (from consumer end)
Ref -> https://react-bootstrap.github.io/components/cards/#basic-example